### PR TITLE
binary conventions for sparse arrays

### PR DESCRIPTION
**Pull request by ahwillia**
As discussed here:

https://github.com/JuliaML/LossFunctions.jl/issues/56

Ready to merge after review by @Evizero

### CODE REVIEW

File: test/tst_loss.jl [hunk 1]
```diff
@@ -393,3 +393,42 @@ end
...
+        sparse_target[nz] = 1.0
+        @test typeof(sparse_target) <: AbstractSparseArray
+
+        # dense vector of {-1,1}
+        target = [ t == 0.0 ? -1.0 : 1.0 for t in sparse_target ]
+        
+        output = randn(N)
+
+        for loss in margin_losses
+            @test isapprox(value(loss,sparse_target,output), value(loss,target,output))
```
Comment (Evizero) [inline test/tst_loss.jl]: Oh, yea. Looking at the test failures reminded me. For our test to work in 0.6 we need to escape the `value` to `LossFunctions.value` at the moment.

Comment (ahwillia) [inline test/tst_loss.jl]: Huh. Will do. Everything passes for me on v0.5 so hopefully this is the only hiccup.

**Comment by ahwillia** [comment]
Do you think it is a good idea to be testing against `v0.6` this early? I'm inclined to not do that at this time... but I'll leave that decision to you.

**Comment by Evizero** [comment]
Well, as long as the differences are manageable I kinda like the excuse to stay aware of what needs to change down the line. That said, once it is annoying enough I will surely switch it off

**Pull request merged**